app/models/school.rb

```diff
@@ -41,15 +41,15 @@ class School < ApplicationRecord
   }
 
   scope :partnered, lambda { |year|
-    where(id: Partnership.joins(:cohort).where(cohorts: { start_year: year }).pluck(:school_id))
+    where(id: Partnership.joins(:cohort).where(cohorts: { start_year: year }).select(:school_id))
```

Comment:
Why?

Reply:
Not-related minor improvement. `where(id: other_scope.pluck(:id)` fires two sql queries. First one `SELECT id FROM <subscope>` and second `SELECT * FROM <scope> WHERE id IN (<massive list of ids fetched from first qury>)`. Using `select` AR performs a single query with subquery: `SELECT * FROM <scope> WHERE id in (SELECT id from <subscope>)`. I had some really nasty experience with AR cache memory leaking due to caching every single variant of `WHERE id in (<list of ids>)`.

Reply:
Nice, good to know. Thanks!